libraries/AC_Autorotation/AC_Autorotation.cpp

```diff
@@ -219,10 +219,9 @@ float AC_Autorotation::get_rpm(bool update_counter)
 
         //Get RPM value
         uint8_t instance = _param_rpm_instance;
-        current_rpm = rpm->get_rpm(instance);
 
         //Check RPM sesnor is returning a healthy status
-        if (current_rpm <= -1) {
+        if (!rpm->get_rpm(instance, current_rpm) || current_rpm <= -1) {
```

Comment:
Shouldn't the `current_rpm <= -1` be removed here? This was the old failure reporting, but you've moved that to a boolean now.

Reply:
the health check doesn't specifically check for -1, so to ensure we maintain current semantics so I don't need to retest autorotation code I want to keep it in